### DISCUSSION TITLE
Sending the audio output of another program to Jamulus (MacOS & Jack)

### DISCUSSION THREAD

**Discussion by adediego** [jamulussoftware/jamulus#1633] [Technical Support & Help] [answered]
I am using Jamulus on MacOS and it works great with the native audio but I want to pipe in the audio output of another program (e.g. a metronome) directly and I didn't see how to do that without Jack. 

So I installed Jack and QJackCtl and they start without problems and the jack server is recognized by e.g. Ardour. 

However, in the list of possible devices in the Jamulus settings, there is no entry "jack" and Jamulus doesn't appear in the Graph connection window in QJackCtl (it does under Linux).

Did somebody get this running in MacOS or has an alternative solution for piping the audio output of another program into Jamulus?

**Comment by hoffie** [comment, marked as answer]
Not a Mac user myself, but I seem to remember others to mention "Blackhole" for audio routing on Macs. Could this be an option?

If you want to use Jamulus with jack on Mac, you'll likely need a custom build which was built with `qmake CONFIG+=jackonmac && make` as the default build uses coreaudio.

**Reply by dcorson-ticino-com** [reply]
I would also suggest taking a look at the doc on Jamulus.io
There is a lot of info about such special uses.

**Reply by adediego** [reply]
I built Jamulus with ```qmake CONFIG+=jackonmac``` and it works like a charm. Thanks a lot!
Blackhole seems nice too, good to know about it. Jack works for what I want to do though, so I'll stick with it for now. 

@dcorson-ticino-com I did look through the docs but I missed the reference to Blackhole under "Using Jamulus audio in Zoom (or other) meeting apps". My bad. The option ```jackonmac``` doesn't seem to be mentioned anywhere else than in the ChangeLog though.

**Reply by hoffie** [reply]
> The option jackonmac doesn't seem to be mentioned anywhere else than in the ChangeLog though.

Certainly possible. Maybe you could [send a Pull Request](https://github.com/jamulussoftware/jamuluswebsite/blob/release/CONTRIBUTING.md) for adding it to the Tipps and Tricks section?

**Comment by ewarning** [comment]
I use Loopback with a MacBbok Pro running Big Sur and it works like a charm! Loopback is a paid program though.
